mitiq/typing.py

```diff
@@ -115,11 +124,47 @@ def nqubits(self) -> int:
     def asarray(self) -> npt.NDArray[np.int64]:
         return self._bitstrings
 
-    def __getitem__(self, indices: List[int]) -> npt.NDArray[np.int64]:
-        return np.array([self._measurements[i] for i in indices]).T
-
-    def __iter__(self) -> Iterable[Bitstring]:
-        yield from self.result
+    @classmethod
+    def from_counts(
+        cls,
+        counts: Dict[str, int],
+        qubit_indices: Optional[Tuple[int, ...]] = None,
+    ) -> "MeasurementResult":
+        """Initializes a MeasurementResult from a dictionary of counts.
+
+        Example:
+        >>> MeasurementResult.from_counts({"00": 175, "11": 177})
+        """
+        counter = Counter(counts)
+        return cls(list(counter.elements()), qubit_indices)
+
+    def get_counts(self) -> Dict[str, int]:
+        """Returns a colleciton.Counter whose keys are the measured
```

Comment:
```suggestion
        """Returns a dictionary whose keys are the measured
```

Reply:
The following line also needs a tweak as well FYI.